plugins/scaffolder-react/src/hooks/useCustomFieldExtensions.ts

```diff
@@ -33,10 +33,9 @@ export const useCustomFieldExtensions = <
   outlet: React.ReactNode,
 ) => {
   // Get custom fields created with FormFieldBlueprint
-  const apiHolder = useApiHolder();
-  const formFieldsApi = apiHolder.get(formFieldsApiRef);
+  const formFieldsApi = useApi(formFieldsApiRef);
   const [{ result: blueprintFields }, methods] = useAsync(
-    formFieldsApi?.getFormFields ?? (async () => []),
+    formFieldsApi?.getFormFields,
```

Comment:
Now you can remove the question mark. But wasn't that intentional, that the api holder was used so that the API was explicitly optional to provide?

Reply:
Na, there's a default implementation shipped with the `plugin` anyways, which just returns `[]`.

In the new frontend system it will automatically add things to that array, but in the old system for backwards compat it just returns empty. You can of course move away from `JSX` and use that API if you want and provide your own implementation, but it's probably not worth it.